tests/channel_internal_ut/channel_internal_ut.c

```diff
@@ -0,0 +1,774 @@
+
+    //act
+    real_async_op_cancel(pull_op);
+    work_function(work_context);
+
+    //assert
+    ASSERT_ARE_EQUAL(int, 1, pull_context);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    //cleanup
+    THANDLE_ASSIGN(ASYNC_OP)(&pull_op, NULL);
+    THANDLE_ASSIGN(CHANNEL_INTERNAL)(&channel_internal, NULL);
+}
+
+
+/*Tests_SRS_CHANNEL_INTERNAL_43_134: [ cancel_op shall call srw_lock_acquire_exclusive. ]*/
+/*Tests_SRS_CHANNEL_INTERNAL_43_146: [ If the operation is not in the list of pending operations and the result of the operation is CHANNEL_CALLBACK_RESULT_OK, cancel_op shall set the result of the operation to CHANNEL_CALLBACK_RESULT_CANCELLED. ]*/
+/*Tests_SRS_CHANNEL_INTERNAL_43_139: [ cancel_op shall call srw_lock_release_exclusive. ]*/
+TEST_FUNCTION(cancel_op_cancels_matched_op)
+{
+    //arrange
+    THANDLE(CHANNEL_INTERNAL) channel_internal = channel_internal_create_and_open(test_threadpool.test_threadpool);
+    THANDLE(RC_PTR) data = rc_ptr_create_with_move_pointer(test_data, test_data_dispose);
+    umock_c_reset_all_calls();
+
+    THREADPOOL_WORK_FUNCTION work_function;
+    void* work_context;
+    setup_first_op_expectations();
+    setup_second_op_expectations(false, &work_function, &work_context);
+
+    int32_t pull_context = 0;
+    THANDLE(ASYNC_OP) pull_op = NULL;
+    int32_t push_context = 0;
+    THANDLE(ASYNC_OP) push_op = NULL;
+
+    CHANNEL_RESULT pull_result = channel_internal_pull(channel_internal, test_pull_callback_matched_cancelled, &pull_context, &pull_op);
+    CHANNEL_RESULT push_result = channel_internal_push(channel_internal, data, test_push_callback_cancelled, &push_context, &push_op);
+
+    ASSERT_IS_NOT_NULL(pull_op);
+    ASSERT_IS_NOT_NULL(push_op);
+    ASSERT_ARE_EQUAL(CHANNEL_RESULT, CHANNEL_RESULT_OK, pull_result);
+    ASSERT_ARE_EQUAL(CHANNEL_RESULT, CHANNEL_RESULT_OK, push_result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    umock_c_reset_all_calls();
+
+    STRICT_EXPECTED_CALL(srw_lock_acquire_exclusive(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(srw_lock_release_exclusive(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(THANDLE_ASSIGN(RC_PTR)(IGNORED_ARG, NULL));
+    STRICT_EXPECTED_CALL(THANDLE_INITIALIZE_MOVE(ASYNC_OP)(IGNORED_ARG, IGNORED_ARG));
+    STRICT_EXPECTED_CALL(THANDLE_ASSIGN(ASYNC_OP)(IGNORED_ARG, NULL));
+
+    //act
+    real_async_op_cancel(pull_op);
+    work_function(work_context);
+
+    //assert
+    ASSERT_ARE_EQUAL(CHANNEL_RESULT, CHANNEL_RESULT_OK, pull_result);
+    ASSERT_ARE_EQUAL(int, 1, pull_context);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    //cleanup
+    THANDLE_ASSIGN(ASYNC_OP)(&push_op, NULL);
+    THANDLE_ASSIGN(ASYNC_OP)(&pull_op, NULL);
+    THANDLE_ASSIGN(RC_PTR)(&data, NULL);
+    THANDLE_ASSIGN(CHANNEL_INTERNAL)(&channel_internal, NULL);
+}
+
+END_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
```

Comment:
>TEST_SUITE_NAME_FROM_CMAKE [](http://example.com/codeflow?start=15&length=26)

I've briefly ran the code coverage for these and they are not touching all the lines of code.

Here's SOME lines that are not exercised from channel_internal.c: 387, 322, 276, 80.